src/transformers/convert_slow_tokenizer.py

```diff
@@ -429,6 +430,12 @@ def __init__(self, *args):
             m.ParseFromString(f.read())
         self.proto = m
 
+        if self.proto.trainer_spec.byte_fallback:
+            warnings.warn(
+                "The sentencepiece tokenizer that you are converting to a fast tokenizer uses the byte fallback option"
+                " which is not implemented in the fast tokenizers."
```

Comment:
Could the warning explain a bit more to the user what to expect? If I read that message with no context, I have no idea what it means 😬

Reply:
Also should we maybe use the `logger` here instead:

```py
from .utils import logger
...

logger = logging.get_logger(__name__)

....

logger.warn(...)
```

Reply:
@sgugger (@patrickvonplaten ) That's a good point! I just added an explanation in my last commit. Does this seem sufficient? :slightly_smiling_face:

Reply:
@patrickvonplaten That's actually how I did it [initially](https://github.com/huggingface/transformers/pull/16629/commits/1ee9f3d962a9290131b74efb41725384cc33faf7) but then I saw that the python `warnings` object was also used elsewhere in the transformers codebase so I changed it back. It seemed to me that `warnings.warn` was a warning that would probably be more read by the user. 

I have no preference between the two, but I'm interested to know what makes you prefer the second option! :smile:

Reply:
From the official docs:

```


Issue a warning regarding a particular runtime event | warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted
-- | --


Issue a warning regarding a particular runtime event
	

[warnings.warn()](https://docs.python.org/3.8/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning

[logging.warning()](https://docs.python.org/3.8/library/logging.html#logging.warning) if there is nothing the client application can do about the situation, but the event should still be noted
```
https://docs.python.org/3.8/howto/logging.html#when-to-use-logging

I think this is more of a `warnings` case, where user SHOULD change his code as it's likely to cause issues to him later on.
Not raising Exception here is mainly due to the fact that a alerted user might KNOW he's not going to trigger byte fallback, and wants the offsets.

Reply:
Yes it's definitely not instance where we want to use the `logger` as, from what I understood, the user should always see this and adapt the code.